src/products.js

```diff
@@ -0,0 +1,117 @@
+const { log, saveFiles } = require('cozy-konnector-libs')
+const { rootUrl, request } = require('./request')
+const cozyhelpers = require('./cozyhelpers')
+
+const tableUrl = rootUrl + '/webapp/wcs/stores/controller/ec/products/table'
+const billPath = '/webapp/wcs/stores/controller/OrderBillDisplay'
+const firstPageNum = 1
+
+module.exports = {
+  fetchBills
+}
+
+function fetchBills(folderPath) {
+  return fetchPagesCount()
+    .then(fetchPages)
+    .then(products => fetchBillFiles(products, folderPath))
+}
+
+function fetchPagesCount() {
+  return requestTable(firstPageNum).then(parsePagesCount)
+}
+
+function parsePagesCount($) {
+  const lastPageString = $('a[data-page]')
+    .last()
+    .data('page')
+
+  return lastPageString ? parseInt(lastPageString) : firstPageNum
+}
+
+function fetchPages(pagesCount) {
+  log('info', `Found ${pagesCount} product page(s).`)
+
+  let productsPromise = Promise.resolve([])
+
+  for (let pageNum = firstPageNum; pageNum <= pagesCount; pageNum++) {
+    productsPromise = productsPromise.then(products =>
+      fetchPage(pageNum).then(foundProducts => products.concat(foundProducts))
+    )
+  }
+
+  return productsPromise
+}
+
+function fetchPage(pageNum) {
+  return requestTable(pageNum).then($ => parseTable(pageNum, $))
+}
+
+function requestTable(pageNum) {
+  return request({
+    method: 'GET',
+    url: tableUrl,
+    qs: {
+      filtre: '0',
+      pagination: pageNum.toString()
+      // Third parameter `time` is a timestamp probably matching the current
+      // date according to the configured timezone of the server, e.g.
+      // `+new Date()`. We ignore it since it works without it anyway.
+    }
+  })
+}
+
+function parseTable(pageNum, $) {
+  log('info', `Parsing products page ${pageNum}...`)
+
+  const products = $('.item_product')
```

Comment:
The [scrape](https://github.com/konnectors/libs/blob/master/packages/cozy-konnector-libs/docs/api.md#scrape-specs-childselector--object--array) function may be you friend for this (but not blocking for me)